src/plist.h

```diff
@@ -314,35 +314,54 @@ EXPORT_INLINE Obj NewImmutableEmptyPlist(void)
 /****************************************************************************
 **
 *F  NewPlistFromArray(<list>,<length>) . . create a plain list from a C array
+*F  NewPlistFromArgs(<args...>) .  create a plain list from list of arguments
+**
+**  This macro turns a variable-length list of macro arguments into an array,
+**  which is then passed to NewPlistFromArray.
+**
+**  __VA_ARGS__ contains the list of arguments given to this macro.
+**  There is no (nice) way to convert this list of arguments to an array
+**  in both C and C++, so we need two implementations
 */
-EXPORT_INLINE Obj NewPlistFromArray(const Obj * list, Int length)
+
+#ifdef __cplusplus
+} // escape extern "C"
```

Comment:
Not for this PR, but: wrapping `#includes` in `extern "C"` was a quick&dirty hack I did; on the long run, it'd probably be better if each of our headers did this instead (after its own `#includes`):
```C++
#ifdef __cplusplus
extern "C" {
#endif 
... original content ...
#ifdef __cplusplus
}
#endif 
```

Reply:
Now handling this in a different way (as for some reason this didn't work in cygwin -- not sure why)